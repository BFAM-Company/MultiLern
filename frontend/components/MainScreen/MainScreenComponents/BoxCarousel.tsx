import React from 'react';
import { Animated, Dimensions, ScrollView } from 'react-native';
import { View, Text, SafeAreaView, StyleSheet, ImageBackground, Image, TouchableOpacity, Platform } from 'react-native';
import FeatureBox from './FeatureBox';



function BoxCarousel({pageSwitcher}: any) {
  const IMAGES = {
    ImageBackground1: require('./../../../assets/cool-background.png'),
    ImageBackground2: require('./../../../assets/cool-background (1).png'),
    ImageBackground3: require('./../../../assets/cool-background (2).png'),
    ImageBackground4: require('./../../../assets/cool-background (3).png'),
  }

  return (
    <View style={styles.mainContainer}>
        <ScrollView
            style={styles.scroll}
            horizontal={true}
        >   
            <FeatureBox buttonAction={()=>{pageSwitcher('main')}} title={'Rozwiązania Zadań'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground1}>
                Najlepsze rozwiązania zadań. Wszystkie podręcziki i 
                sprawdzone odpowiedzi przez ekspertów i specjalną sztuczną inteligencję
            </FeatureBox>
            <FeatureBox buttonAction={()=>{pageSwitcher('main')}} title={'Nowoczesne Sposoby Nauki'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground2}>
                Już dziś zacznij korzystać z nowoczesnych technik nauki i pomocy naukowych. 
                Fiszki, notatki to tylko część dostępnych narzędzi
            </FeatureBox>
            <FeatureBox buttonAction={()=>{pageSwitcher('main')}} title={'Baza Sprawdzianów'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground3}>
                Największa baza sprawdzianów, z różnych szkół, nauczycieli i poziomów nauczania. 
                Sprawdź czy mamy coś co może Ci pomóc
            </FeatureBox>
            <FeatureBox buttonAction={()=>{pageSwitcher('main')}} title={'Korepetycje'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground4}>
                Znajdź osoby chętne do pomocy z całej Polski. Umów spotakania za pośrednictwem naszego chatu. Daj sobie pomóc!
            </FeatureBox>
        </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
    mainContainer:{
        width:'100%',
        marginTop:50,
    },
    scroll:{
        width:'100%',
        height:'auto',
        paddingBottom:20,
        paddingTop:20,
        backgroundColor:'#fff'
    },
})


export default BoxCarousel;