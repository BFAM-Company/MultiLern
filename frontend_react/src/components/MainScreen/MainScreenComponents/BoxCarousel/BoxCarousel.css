.mainContainer {
  width: 100%;
  margin-top: 50px;
}

.scroll {
  width: 100%;
  padding: 20px 0;
  background-color: #fff;
}

.carousel {
  display: flex;
  overflow-x: auto;
  gap: 20px;
  padding-bottom: 20px;
}

.carousel::-webkit-scrollbar {
  height: 8px;
}

.carousel::-webkit-scrollbar-thumb {
  background-color: #888;
  border-radius: 10px;
}

.carousel::-webkit-scrollbar-thumb:hover {
  background-color: #555;
}
