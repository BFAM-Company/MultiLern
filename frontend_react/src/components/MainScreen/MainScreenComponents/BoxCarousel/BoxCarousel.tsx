import React from 'react';
import { useState } from 'react';
import './BoxCarousel.css'; // Assuming you'll have a separate CSS file for styles
import FeatureBox from '../FeatureBox/FeatureBox.tsx'; // Assuming you have a similar FeatureBox component for the web
import { useNavigate } from 'react-router-dom';

function BoxCarousel() {
    const navigate = useNavigate()
  const IMAGES = {
    ImageBackground1: '/assets/cool-background.png',
    ImageBackground2: '/assets/cool-background (1).png',
    ImageBackground3: '/assets/cool-background (2).png',
    ImageBackground4: '/assets/cool-background (3).png',
  };

  return (
    <div className="mainContainer">
      <div className="scroll">
        <div className="carousel">
          <FeatureBox buttonAction={() => navigate('/main')} title={'Rozwiązania Zadań'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground1}>
            Najlepsze rozwiązania zadań. Wszystkie podręczniki i sprawdzone odpowiedzi przez ekspertów i specjalną sztuczną inteligencję.
          </FeatureBox>
          <FeatureBox buttonAction={() => navigate('/main')} title={'Nowoczesne Sposoby Nauki'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground2}>
            Już dziś zacznij korzystać z nowoczesnych technik nauki i pomocy naukowych. Fiszki, notatki to tylko część dostępnych narzędzi.
          </FeatureBox>
          <FeatureBox buttonAction={() => navigate('/main')} title={'Baza Sprawdzianów'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground3}>
            Największa baza sprawdzianów, z różnych szkół, nauczycieli i poziomów nauczania. Sprawdź czy mamy coś, co może Ci pomóc.
          </FeatureBox>
          <FeatureBox buttonAction={() => navigate('/main')} title={'Korepetycje'} buttonText={'Zobacz'} imageSource={IMAGES.ImageBackground4}>
            Znajdź osoby chętne do pomocy z całej Polski. Umów spotkania za pośrednictwem naszego chatu. Daj sobie pomóc!
          </FeatureBox>
        </div>
      </div>
    </div>
  );
}

export default BoxCarousel;
