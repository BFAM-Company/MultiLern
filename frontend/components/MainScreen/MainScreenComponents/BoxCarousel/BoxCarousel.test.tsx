import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react-native';
import BoxCarousel from './BoxCarousel';

describe('BoxCarousel', () => {
  afterEach(cleanup);

  it('renders correctly', () => {
    const mockPageSwitcher = jest.fn();
    const { toJSON } = render(<BoxCarousel pageSwitcher={mockPageSwitcher} />);
    expect(toJSON()).toMatchSnapshot();
  });
   it('renders all feature boxes correctly', () => {
    const mockPageSwitcher = jest.fn();
    const { queryAllByTestId} = render(<BoxCarousel pageSwitcher={mockPageSwitcher} />);
    
    const featureBoxes = queryAllByTestId('feature-box');

    expect(featureBoxes).toHaveLength(4);
  });

  it('calls pageSwitcher function when button in FeatureBox is pressed', () => {
    const mockPageSwitcher = jest.fn();
    const { queryAllByText } = render(<BoxCarousel pageSwitcher={mockPageSwitcher} />);
    
    const buttons = queryAllByText('Zobacz');
    for(let i = 0; i < buttons.length; i++) {
        const button = buttons[i];
        fireEvent.press(button);
        expect(mockPageSwitcher).toHaveBeenCalledTimes(i + 1);
        expect(mockPageSwitcher).toHaveBeenCalledWith('main');
    }
    
  });
});
