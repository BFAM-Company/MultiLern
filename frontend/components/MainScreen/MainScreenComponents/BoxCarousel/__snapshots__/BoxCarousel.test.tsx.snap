// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxCarousel renders correctly 1`] = `
<View
  style={
    {
      "marginTop": 50,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    horizontal={true}
    style={
      {
        "backgroundColor": "#fff",
        "height": "auto",
        "paddingBottom": 20,
        "paddingTop": 20,
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "elevation": 16,
            "height": 250,
            "marginBottom": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.44,
            "shadowRadius": 10.32,
            "width": 320,
          }
        }
        testID="feature-box"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "alignItems": "flex-start",
              "borderRadius": 10,
              "display": "flex",
              "height": "100%",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": "100%",
                  "width": "100%",
                },
                {
                  "borderRadius": 10,
                },
              ]
            }
          />
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 24,
                "fontWeight": "900",
                "marginLeft": 10,
              }
            }
          >
            Rozwiązania Zadań
          </Text>
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 18,
                "marginLeft": 10,
              }
            }
          >
            Najlepsze rozwiązania zadań. Wszystkie podręcziki i sprawdzone odpowiedzi przez ekspertów i specjalną sztuczną inteligencję
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "rgb(33,33,43)",
                "borderRadius": 10,
                "display": "flex",
                "height": 50,
                "justifyContent": "center",
                "marginLeft": 10,
                "opacity": 1,
                "width": 120,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 20,
                  "fontWeight": "700",
                }
              }
            >
              Zobacz
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "elevation": 16,
            "height": 250,
            "marginBottom": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.44,
            "shadowRadius": 10.32,
            "width": 320,
          }
        }
        testID="feature-box"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "alignItems": "flex-start",
              "borderRadius": 10,
              "display": "flex",
              "height": "100%",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": "100%",
                  "width": "100%",
                },
                {
                  "borderRadius": 10,
                },
              ]
            }
          />
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 24,
                "fontWeight": "900",
                "marginLeft": 10,
              }
            }
          >
            Nowoczesne Sposoby Nauki
          </Text>
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 18,
                "marginLeft": 10,
              }
            }
          >
            Już dziś zacznij korzystać z nowoczesnych technik nauki i pomocy naukowych. Fiszki, notatki to tylko część dostępnych narzędzi
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "rgb(33,33,43)",
                "borderRadius": 10,
                "display": "flex",
                "height": 50,
                "justifyContent": "center",
                "marginLeft": 10,
                "opacity": 1,
                "width": 120,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 20,
                  "fontWeight": "700",
                }
              }
            >
              Zobacz
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "elevation": 16,
            "height": 250,
            "marginBottom": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.44,
            "shadowRadius": 10.32,
            "width": 320,
          }
        }
        testID="feature-box"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "alignItems": "flex-start",
              "borderRadius": 10,
              "display": "flex",
              "height": "100%",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": "100%",
                  "width": "100%",
                },
                {
                  "borderRadius": 10,
                },
              ]
            }
          />
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 24,
                "fontWeight": "900",
                "marginLeft": 10,
              }
            }
          >
            Baza Sprawdzianów
          </Text>
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 18,
                "marginLeft": 10,
              }
            }
          >
            Największa baza sprawdzianów, z różnych szkół, nauczycieli i poziomów nauczania. Sprawdź czy mamy coś co może Ci pomóc
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "rgb(33,33,43)",
                "borderRadius": 10,
                "display": "flex",
                "height": 50,
                "justifyContent": "center",
                "marginLeft": 10,
                "opacity": 1,
                "width": 120,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 20,
                  "fontWeight": "700",
                }
              }
            >
              Zobacz
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "elevation": 16,
            "height": 250,
            "marginBottom": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.44,
            "shadowRadius": 10.32,
            "width": 320,
          }
        }
        testID="feature-box"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "alignItems": "flex-start",
              "borderRadius": 10,
              "display": "flex",
              "height": "100%",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": "100%",
                  "width": "100%",
                },
                {
                  "borderRadius": 10,
                },
              ]
            }
          />
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 24,
                "fontWeight": "900",
                "marginLeft": 10,
              }
            }
          >
            Korepetycje
          </Text>
          <Text
            style={
              {
                "color": "rgb(33,33,43)",
                "fontSize": 18,
                "marginLeft": 10,
              }
            }
          >
            Znajdź osoby chętne do pomocy z całej Polski. Umów spotakania za pośrednictwem naszego chatu. Daj sobie pomóc!
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "rgb(33,33,43)",
                "borderRadius": 10,
                "display": "flex",
                "height": 50,
                "justifyContent": "center",
                "marginLeft": 10,
                "opacity": 1,
                "width": 120,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 20,
                  "fontWeight": "700",
                }
              }
            >
              Zobacz
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
